src/lazyapplying.jl

```diff
@@ -84,10 +84,30 @@ similar(bc::Broadcasted{ApplyArrayBroadcastStyle{N}}, ::Type{ElType}) where {N,E
     copyto!(dest, first(bc.args))
 end
 @inline function copyto!(dest::AbstractArray, bc::Broadcasted{ApplyArrayBroadcastStyle{N}}) where N 
-    @assert length(bc.args) == 1
-    copyto!(dest, first(bc.args))    
+    if length(bc.args) == 1
+        copyto!(dest, first(bc.args))
+        if bc.f !== identity
+            dest .= bc.f.(dest)
```

Comment:
This is a bit dangerous definition considering input and output types of `bc.f` can be different.